src/main/java/it/pagopa/interop/probing/eservice/registry/reader/config/PropertiesLoader.java

```diff
@@ -0,0 +1,49 @@
+/**************************************************************************
+*
+* Copyright 2023 (C) DXC
+*
+* Created on  : Feb 24, 2023
+* Author      : dxc technology
+* Project Name: interop-be-probing-eservice-registry-reader 
+* Package     : it.pagopa.interop.probing.eservice.registry.reader.config
+* File Name   : PropertiesLoader.java
+*
+*-----------------------------------------------------------------------------
+* Revision History (Release )
+*-----------------------------------------------------------------------------
+* VERSION     DESCRIPTION OF CHANGE
+*-----------------------------------------------------------------------------
+** --/1.0  |  Initial Create.
+**---------|------------------------------------------------------------------
+***************************************************************************/
+package it.pagopa.interop.probing.eservice.registry.reader.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * The Class PropertiesLoader.
+ */
+public class PropertiesLoader {
+
+	private PropertiesLoader() {
+		super();
+	}
+
+	/**
+	 * Load properties.
+	 *
+	 * @param resourceFileName the resource file name
+	 * @return the properties
+	 * @throws IOException Signals that an I/O exception has occurred.
+	 */
+	public static Properties loadProperties(String resourceFileName) throws IOException {
+		Properties configuration = new Properties();
```

Comment:
Use Try with resources

Reply:
What do you expect in the catch clause? Because i can add a log that specifies that an error occured, but then i need to re-throw the exception , otherwise the execution will continue until one component will try to utilize the properties extracted.

Reply:
Ale, loadProperties has to be call once at startup and pass Properties as a dependency or however the first time load from file and next times pass only the Properties. You can use a property library such as https://github.com/lightbend/config for this scope. InputStream is autoClosable and to avoid to close by your own in a finally clause. You can throw exception and log in the same time what happened.

Reply:
We implemented the singleton on the propertyLoader so only the first time they are loaded, then the following times we use the same instance.